### PR TITLE
prov/shm: address name fixes

### PR DESCRIPTION
**Pull request by aingerson**
2 address name fixes:
- include NULL terminator in addrlen to make sure to duplicate/use it
- cast address correctly in fi_av_insert

### CODE REVIEW

**Comment by aingerson** [comment]
Updated to include same fix in getname path

File: prov/shm/src/smr_ep.c [hunk 1]
```diff
@@ -61,17 +61,19 @@ int smr_setname(fid_t fid, void *addr, size_t addrlen)
...
 	ep = container_of(fid, struct smr_ep, util_ep.ep_fid.fid);
 	if (!ep->name)
 		return -FI_EADDRNOTAVAIL;
 
 	if (!addr || *addrlen == 0 ||
 	    snprintf(addr, *addrlen, "%s", ep->name) >= *addrlen)
-		ret = -FI_ETOOSMALL;
-	*addrlen = strlen(ep->name);
-	return ret;
+		return -FI_ETOOSMALL;
```
Comment (shefty) [inline prov/shm/src/smr_ep.c]: You need to set addrlen to the required length prior to returning.  This tells the app what size it needs to allocate.

Comment (aingerson) [reply prov/shm/src/smr_ep.c]: Ahh yes, that makes more sense to why that was the way it was. Updated.

**Comment by aingerson** [comment]
Updated again and included second patch related to addressing. This should fix #5465

**Comment by aingerson** [comment]
@shefty osx travis failure looks like a fluke. Thoughts on merging this?

**Comment by shefty** [comment]
What are the jenkins failures?  Changes themselves look fine.

**Comment by aingerson** [comment]
Mellanox failures but I think it happened when we restated the SM so I don't think they're related. I can retest to be sure?

**Pull request merged**